movement_primitives/dmp/_base.py

```diff
@@ -16,23 +16,33 @@ def reset(self):
         self.current_y = np.copy(self.start_y)
         self.current_yd = np.copy(self.start_yd)
 
+
+class WeightParametersMixin:
+    """Mixin class providing common access methods to forcing term weights."""
     def get_weights(self):
         """Get weight vector of DMP.
 
         Returns
         -------
         weights : array, shape (N * n_weights_per_dim,)
-            Current weights of the DMP. N depend on the type of DMP (task space dim. or for dual-cartesian=12)
+            Current weights of the DMP. N depends on the type of DMP
```

Comment:
This does not necessarily have to be a DMP, could also be a ProMP!

Reply:
On the other hand, the default implementation uses self.forcing_term, which is very specific for DMPs... I don't know what to do here.

Reply:
more general terms could be movement_primitive, movement class, representation or behavior.

Reply:
yep, but in that case we would have to move this to movement_primitive.base and then we have DMP-specific code there...